lib/elixir/lib/calendar.ex

```diff
@@ -1616,6 +1890,18 @@ defmodule DateTime do
     System.convert_time_unit((seconds - @unix_epoch) * 1_000_000 + microsecond, :microsecond, unit)
   end
 
+  def to_unix(%DateTime{calendar: Calendar.ISO, year: year}, _unit) when year < 0 do
+    raise ArgumentError
```

Comment:
This is actually very interesting: This behaviour was already there (I believe it resulted in a MatchError before), but I had to extract it to a separate clause to make the function work in a more general fashion.

I have no idea _why_ this behaviour is the way it is. Erlang's `:calendar.datetime_to_gregorian_seconds` does not accept years <= `0`, but it does not state why.